util/deploy-production.sh

```diff
@@ -143,3 +136,32 @@ fi
 # Update and close deployment bug.
 LAST_DEPLOYMENT_ISSUE=$(gh issue list --state open --label "$PROD_LABEL" --label "$RELEASE_LABEL" --limit 1 --json number --jq '.[] | .number')
 gh issue close "$LAST_DEPLOYMENT_ISSUE" -c "Deployment is now complete."
+
+# Check if there are more more than two versions of the default service left
+# after we're done with this deplyment to make sure there's room for the next
+# deployment. If there are, ask to delete the oldest default service version,
+# and also delete the same version from the other services which will also exist
+# if all went well during the deployment. This check isn't fail safe, but
+# combined with the check we do before doing any deployments earlier in this
+# script, this should leave us in a good state.
+
+VERSIONS=$(gcloud app --project=wptdashboard versions list --filter="service=default" --format=list | wc -l)
+
+if (($VERSIONS == 3)); then
+  echo -e "Please ensure the deployment was successful. If so, we can go ahead and\ndelete the oldest version of all services if necessary, leaving the one just\ndeployed and the one running before this deployment. This will ensure we leave\nroom for the next deployment.\n"
+
+  read -p "Delete oldest version of all services to leave room for the next deplyment? (y/n): " DELETE
+
+  if [[ $DELETE == "y" ]]; then
+    echo "Found $VERSIONS for the default service, deleting the oldest version of all services."
+
+    OLDEST_REV=$(gcloud app --project=wptdashboard versions list --sort-by=last_deployed_time --filter="service=$SERVICE" --limit=1 --format=json | jq -r '.[] | .id')
+    for SERVICE in $SERVICES; do
+      echo "Deleting $SERVICE service version $OLDEST_REV"
+      gcloud app --project=wptdashboard versions delete --service=$SERVICE --quiet $OLDEST_REV
+    done
+  elif (($VERSIONS > 3)); then
```

Comment:
Currently, this branch is nested inside the `VERSIONS == 3` branch. This should be unnested to be at the same level:

```
if (($VERSIONS == 3)); then
  ..
elif (($VERSIONS > 3)); then
..
fi
```